src/EntryPoints/PageApprovalsHooks.php

```diff
@@ -17,4 +20,46 @@ public static function onOutputPageParserOutput( OutputPage $out, ParserOutput $
 		);
 	}
 
+	public static function onOutputPageBeforeHTML( OutputPage $out ): void {
+		$user = $out->getUser();
+		$title = $out->getTitle();
+
+		if ( !$title instanceof Title ) {
+			return;
+		}
+
+		$permissionManager = MediaWikiServices::getInstance()->getPermissionManager();
+		$revisionStore = MediaWikiServices::getInstance()->getRevisionStore();
+
+		try {
+			$firstRevision = $revisionStore->getFirstRevision( $title );
+			$firstUserId = $firstRevision ? $firstRevision->getUser( 0 ) : null;
+		} catch ( DBError $e ) {
+			wfLogWarning( "Database error fetching first revision: " . $e->getMessage() );
+			return;
+		}
```

Comment:
This function is too big. One thing that can be extracted is getting the firstUserId. Then again, I don't understand why that ID is needed or what you are trying to do.

Reply:
A lot of stuff here is gonna be rewritten, I am only trying to check if user is an approver and if user can approve the page. Since now we have table in the DBs, I can actually write the exact code instead of the placeholder. I'll do it tomorrow.

Reply:
Would be nice to keep the PR focused on the UI. Using placeholder is fine. Figuring out how to get the data to the UI is a separate task.

Reply:
Okay. I''ve done that I've hardcoded all Booleans to keep the code clean. And we can build logic in the scope of other PRs.